src/libsync/owncloudpropagator.cpp

```diff
@@ -618,30 +620,34 @@ bool PropagateDirectory::scheduleNextJob()
         return false;
     }
 
-    // cache the value of first unfinished subjob
     bool stopAtDirectory = false;
-    int i = _firstUnfinishedSubJob;
-    int subJobsCount = _subJobs.count();
-    while (i < subJobsCount && _subJobs.at(i)->_state == Finished) {
-      _firstUnfinishedSubJob = ++i;
-    }
 
-    for (int i = _firstUnfinishedSubJob; i < subJobsCount; ++i) {
-        if (_subJobs.at(i)->_state == Finished) {
+    QMutableVectorIterator<PropagatorJob *> subJobsIterator(_subJobs);
+
+    while (subJobsIterator.hasNext()) {
+        subJobsIterator.next();
+        // Get the state of the state of the sub job pointed by call next()
+        // Function value() will directly access the item through hash in the QList at that subjob
+        if (subJobsIterator.value()->_state == Finished) {
+            // If this items is finish, remove it from the _subJobs list as it is not needed anymore
+            // Note that in this case remove() from QVector will just perform memmove of pointer array items.
+            PropagatorJob * jobPointer = subJobsIterator.value();
+            subJobsIterator.remove();
+            delete jobPointer;
             continue;
```

Comment:
This is a moment in which I am removing the job from the queue and deleting the referenced object since it is not used anymore nowhere.